packages/yarnpkg-fslib/sources/xfs.ts

```diff
@@ -44,19 +43,24 @@ function registerCleanExit() {
   });
 }
 
+let tmpdir: PortablePath | null = null;
+let realTmpdir: PortablePath | null = null;
+
 export const xfs: XFS = Object.assign(new NodeFS(), {
   detachTemp(p: PortablePath) {
     tmpdirs.delete(p);
   },
 
   mktempSync<T>(this: XFS, cb?: (p: PortablePath) => T) {
     registerCleanExit();
+    tmpdir ??= npath.toPortablePath(os.tmpdir());
+    realTmpdir ??= this.realpathSync(tmpdir);
```

Comment:
The `tmpdir` and `realTmpdir` are initialized in `mktempSync` and `mktempPromise` before using.
Convert them to `const` instead of declaring as let on lines 46/47?
```suggestion
    const tmpdir ??= npath.toPortablePath(os.tmpdir());
    const realTmpdir ??= this.realpathSync(tmpdir);
```

Ditto on lines 96/97

Reply:
They're declared outside the scope of the function so it only needs to do the work once.